lib/cocoapods/sources_manager.rb

```diff
@@ -418,6 +438,24 @@ def master_repo_functional?
 
       private
 
+      # @return [Source::Aggregate] The aggregate of the sources from repos.
+      #
+      # @param  [Array<Pathname>] repos
+      #         The local file paths to one or more podspec repo caches.
+      #
+      def aggregate_with_repos(repos)
+        @aggregates_by_repos ||= {}
+
+        # Use the cached version if it exists.
+        if result = @aggregates_by_repos[repos]
```

Comment:
all this can be replaced with `@aggregates_by_repos[repos] ||= Source::Aggregate.new(repos)`